src/main/java/org/sagebionetworks/bridge/spring/filters/MetricsFilter.java

```diff
@@ -77,6 +102,16 @@ private String header(HttpServletRequest request, String name, String defaultVal
         return (value != null) ? value : defaultVal;
     }
 
+    /**
+     * Writes the user's account ID, internal session ID, and app ID to the metrics.
+     * Requires @NotNull for both metrics and session.
+     */
+    private void writeSessionInfoToMetrics(Metrics metrics, UserSession session) {
+        metrics.setSessionId(session.getInternalSessionToken());
+        metrics.setUserId(session.getId());
+        metrics.setAppId(session.getAppId());
+    }
```

Comment:
Do we need this in two places? If we need it in two places, you should create a utility function somewhere that's called in both places instead of copy-pasting the method.